### PR TITLE
Add force-pull flag for dirty repositories

### PR DESCRIPTION
**Pull request by marcosbarbero**
This PR will fix this issue https://github.com/spring-cloud/spring-cloud-config/issues/430

>Note: I didn't find a way to add unit tests for that or even if it can be tested at all. Any suggestion will be appreciate!

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java [hunk 1]
```diff
@@ -20,16 +20,13 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
-import org.eclipse.jgit.api.CheckoutCommand;
-import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.*;
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java]: avoid `*` imports

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java [hunk 2]
```diff
@@ -205,8 +217,33 @@ private Ref checkout(Git git, String label) throws GitAPIException {
 	}
 
 	private boolean shouldPull(Git git, Ref ref) throws GitAPIException {
-		return git.status().call().isClean() && ref != null && git.getRepository()
-				.getConfig().getString("remote", "origin", "url") != null;
+		boolean shouldPull;
+		Status gitStatus = git.status().call();
+		if (this.isForcePull() && !gitStatus.isClean()) {
+            shouldPull = true;
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java]: We use tabs.

Comment (marcosbarbero) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java]: I just forgot to apply the code formatter, sorry about that.

**Comment by spencergibb** [comment]
Squashed and merged with some unit tests via 7b4d0feec408bc6aab78650f1a62f96020aee692

**Comment by andyshinn** [comment]
Can we get an example on how to use this? I don't see any documentation on it.

**Comment by ryanjbaxter** [comment]
@andyshinn please open a separate issue for this.  Maybe @marcosbarbero can continue some documentation

**Comment by marcosbarbero** [comment]
I'll try to work something out today

**Comment by piclemx** [comment]
How to use this forcePull option?

**Comment by spencergibb** [comment]
See the [documentation](https://github.com/spring-cloud/spring-cloud-config/blob/master/docs/src/main/asciidoc/spring-cloud-config.adoc#force-pull-in-git-repositories). It hasn't been released yet.

**Comment by piclemx** [comment]
It's possible to fix this error without adding this option in the cloud config server?

**Comment by spencergibb** [comment]
IIRC, there was something in the original issue

**Comment by marcosbarbero** [comment]
@piclemx here something about it https://cloud.spring.io/spring-cloud-config/spring-cloud-config.html#_version_control_backend_filesystem_use.

In short you need to change the configserver base dir with this property `spring.cloud.config.server.git.basedir`

**Comment by marcosbarbero** [comment]
@spencergibb it has bean released on `Camden.RELEASE`.

**Comment by spencergibb** [comment]
@marcosbarbero my bad, I looked at the wrong commit.

**Comment by spencergibb** [comment]
@piclemx it is in the Camden release train.

**Pull request closed without merging**